crates/edr_napi/src/provider.rs

```diff
@@ -204,16 +210,48 @@ impl Provider {
                 }
             })
             .map_err(|error| napi::Error::new(Status::GenericFailure, error.to_string()))
-            .map(|data| {
+            .and_then(|data| {
                 let solidity_trace = solidity_trace.map(|trace| SolidityTraceData {
                     trace,
                     contract_decoder: Arc::clone(&self.contract_decoder),
                 });
-                Response {
+
+                // Normalise JSON-RPC format for rpcDebugTrace
+                let data = if method_name == "debug_traceTransaction"
+                    || method_name == "debug_traceCall"
+                {
+                    let parsed_data: Result<DebugTraceResult, _> = match &data {
+                        Either::A(json_string) => {
+                            serde_json::from_str::<JsonRpcResponse<DebugTraceResult>>(json_string)
+                                .map(|r| r.result)
+                        }
+                        Either::B(json_value) => serde_json::from_value::<
+                            JsonRpcResponse<DebugTraceResult>,
+                        >(json_value.clone())
+                        .map(|r| r.result),
+                    };
+
+                    match parsed_data {
+                        Ok(trace_result) => {
+                            let transformed = normalise_rpc_debug_trace(trace_result)
+                                .map_err(|e| napi::Error::new(Status::GenericFailure, e))?;
+
+                            let transformed = JsonRpcResponse {
+                                result: transformed,
+                            };
+                            Either::B(serde_json::to_value(transformed)?)
+                        }
+                        Err(_) => data,
+                    }
```

Comment:
Could you please split this into a function with a descriptive name?

Reply:
No need for these lines now after moving normalisation to edr_provider`.